myhdl/_compat.py

```diff
@@ -31,3 +34,7 @@ def to_str(b):
 
     to_bytes = _identity
     to_str = _identity
+
+def ast_parse(s):
+    return compile(s, '<string>', 'exec', \
+                   print_function.compiler_flag|division.compiler_flag|PyCF_ONLY_AST)
```

Comment:
Yes, that’s right, but I do not see any major issue with that because the standard print statement will continue working, and it is taken into account in the visit_Print functions on the _toVerilog and _toVHDL files.

In any case, it is quite possible that I missed something.

> El 26/5/2015, a las 9:19, Keerthan Jaic <notifications@github.com> escribió:
> 
> In myhdl/_compat.py <https://github.com/jandecaluwe/myhdl/pull/81#discussion_r31009197>:
> 
> > @@ -31,3 +34,7 @@ def to_str(b):
> >  
> >      to_bytes = _identity
> >      to_str = _identity
> > +
> > +def ast_parse(s):
> > +    return compile(s, '<string>', 'exec', \
> > +                   print_function.compiler_flag|division.compiler_flag|PyCF_ONLY_AST)
> I think this approach is wrong. This function will always parse the AST with print_function and division enabled even if the particular source file did not import print_function/division from __future__
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/jandecaluwe/myhdl/pull/81/files#r31009197>.
>